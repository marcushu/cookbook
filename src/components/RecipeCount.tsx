import { Box, Typography } from "@mui/material";
import { styled } from "@mui/material/styles";
import { FunctionComponent } from "react";

interface RecipeCountProps {
  numRecipes: Number
  recipeOrFave: string
}

const UserInfo = styled(Box)(({ theme }) => ({
  backgroundColor: 'gray',
  width: '480px',
  [theme.breakpoints.down('sm')]: {
    width: '100%'
  }
}));

const UserInfoText = styled(Typography)(({ theme }) => ({
  color: 'white',
  fontSize: '18px',
  padding: '15px 10px 15px 44px',
  fontWeight: '300',
  [theme.breakpoints.down('sm')]: {
    textAlign: 'center',
    padding: '15px 10px 15px 10px'
  }
}));

const Count = styled('span')({
  color: '#B3F7FF',
  padding: '0px 5px 0px 5px',
  fontSize: '28px',
  verticalAlign: 'middle'
});


const RecipeCount: FunctionComponent<RecipeCountProps> = ({ numRecipes, recipeOrFave }: RecipeCountProps) => {
  return (
    <UserInfo>
      <UserInfoText>
        You currently have
        <Count>{numRecipes}</Count>
        {recipeOrFave}.
      </UserInfoText>
    </UserInfo>
  );
}

export default RecipeCount;