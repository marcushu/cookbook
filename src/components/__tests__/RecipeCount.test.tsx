import { render } from "@testing-library/react"
import RecipeCount from "../RecipeCount"

describe('RecipeCount', () => {
    it('renders', () => {
        const { asFragment } = render(<RecipeCount numRecipes={5} recipeOrFave="Recipes" />);
            expect(asFragment()).toMatchSnapshot();
    });

    it('displays the count and the kind', () => {
        const { getByText } = render(<RecipeCount numRecipes={99} recipeOrFave="recipes" />);

        expect(getByText(/recipes/)).toBeDefined();
        expect(getByText(/99/)).toBeDefined();
    });

})