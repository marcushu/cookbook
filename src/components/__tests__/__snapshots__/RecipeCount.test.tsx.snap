// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCount renders 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-hnu12d"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-174tge4-MuiTypography-root"
    >
      You currently have
      <span
        class="css-vjgswn"
      >
        5
      </span>
      Recipes.
    </p>
  </div>
</DocumentFragment>
`;
